client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/AuthenticationConfigurationBuilder.java

```diff
@@ -173,20 +175,31 @@ public AuthenticationConfigurationBuilder password(char[] password) {
 
    /**
     * Specifies the realm to be used for authentication. Username and password also need to be supplied. If none
-    * is specified, this defaults to 'ApplicationRealm'.
+    * is specified, this defaults to 'default'.
     * Setting this property also implicitly enables authentication (see {@link #enable()}
     */
    public AuthenticationConfigurationBuilder realm(String realm) {
       this.realm = realm;
       return enable();
    }
 
+   public AuthenticationConfigurationBuilder token(String token) {
+      this.token = token;
+      return enable();
+   }
+
    @Override
    public AuthenticationConfiguration create() {
-      String mech = saslMechanism == null ? "DIGEST-MD5" : saslMechanism;
+      String mech = saslMechanism == null ? "SCRAM-SHA-512" : saslMechanism;
```

Comment:
constant string in a single place?

Reply:
Why constify this ?

Reply:
These are SASL mechanism names which don't change. Constifying them makes no sense and would make the examples unreadable.

Reply:
Apologies, I thought you were referencing examples. Fixed.